pkg/controllers/managementuser/rbac/prtb_handler.go

```diff
@@ -152,6 +164,92 @@ func (p *prtbLifecycle) syncPRTB(binding *v3.ProjectRoleTemplateBinding) error {
 	return p.reconcileProjectAccessToGlobalResources(binding, roles)
 }
 
+// ensurePSAPermissions creates the necessary ClusterRole and ClusterRoleBinding
+// to give the 'updatepsa' permission to the project.
+func (p *prtbLifecycle) ensurePSAPermissions(binding *v3.ProjectRoleTemplateBinding, roles map[string]*v3.RoleTemplate) error {
+	// extract project name from the format "cluster-name:project-name"
+	_, projectName, found := strings.Cut(binding.ProjectName, ":")
+	if !found {
+		return fmt.Errorf("invalid project name format")
+	}
+
+	// create AttributesRecord to test if roles allow the updatepsa operation
+	psaRec := authorizer.AttributesRecord{
+		Verb:            "updatepsa",
+		APIGroup:        management.GroupName,
+		Resource:        v32.ProjectResourceName,
+		Name:            binding.ProjectName,
+		ResourceRequest: true,
+	}
+
+	// check if any of the RoleTemplates grant updatepsa permission
+	hasUpdatePSAPermission := false
+	for _, role := range roles {
+		if rbac.RulesAllow(psaRec, role.Rules...) {
+			hasUpdatePSAPermission = true
+		}
+	}
+
+	// only create RBAC resources if user has updatepsa permission
+	if !hasUpdatePSAPermission {
+		return nil
+	}
+
+	// ensure ClusterRole exists with correct updatepsa rules
+	psaCRName := fmt.Sprintf("%s-namespaces-psa", projectName)
+	psaCRWanted := addUpdatepsaClusterRole(projectName)
+	psaCR, err := p.crLister.Get("", psaCRName)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// create ClusterRole if it doesn't exist
+			psaCR, err = p.crClient.Create(psaCRWanted)
+			if err != nil && !apierrors.IsAlreadyExists(err) {
+				return err
+			}
+		} else {
+			return err
+		}
```

Comment:
I slightly prefer early returns, but it's a nit

```suggestion
		if !apierrors.IsNotFound(err) {
			return err
		}
		// create ClusterRole if it doesn't exist
		psaCR, err = p.crClient.Create(psaCRWanted)
		if err != nil && !apierrors.IsAlreadyExists(err) {
			return err
		}
```